src/basic/namespace-util.c

```diff
@@ -551,27 +549,27 @@ int is_idmapping_supported(const char *path) {
         if (r < 0)
                 return r;
 
-        userns_fd = userns_acquire(uid_map, gid_map);
-        if (ERRNO_IS_NEG_NOT_SUPPORTED(userns_fd) || ERRNO_IS_NEG_PRIVILEGE(userns_fd))
+        userns_fd = r = userns_acquire(uid_map, gid_map);
+        if (ERRNO_IS_NEG_NOT_SUPPORTED(r) || ERRNO_IS_NEG_PRIVILEGE(r) || r == -EINVAL)
                 return false;
-        if (userns_fd == -ENOSPC) {
-                log_debug_errno(userns_fd, "Failed to acquire new user namespace, user.max_user_namespaces seems to be exhausted or maybe even zero, assuming ID-mapping is not supported: %m");
+        if (r == -ENOSPC) {
```

Comment:
I don't think duplicating this to `r` is particularly compelling. If the retval is a fd, it makes sense to use that var for error reporting too. We do this everywhere else.

If you really want something like that, please store retval in r alone initially, and assign to fd if the call succeeds

Reply:
"Makes sense" is relative. If the variable is used one or twice, that is reasonable. But with the patch it's repeated three times in the condition and then in the log line and the long variable name adds verbosity that makes the code harder to read. 

Using `r` has the advantage that the error handling paths become more similar, and it's obvious at a glance that the check that is being changed and the one below for `mount_setattr()` are the same.

Also, there is no need to do the store separately. We do that when the original value needs to be preserved on failure. But here we obviously *can* unconditionally assign the store variable, everything is written in a way where it can contain the error value too.